src/navigator/TopBarIconButtonV2.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react'
+import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native'
+import { AnalyticsEventType, AnalyticsPropertiesList } from 'src/analytics/Properties'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import Touchable from 'src/components/Touchable'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+
+interface CommonProps {
+  disabled?: boolean
+  testID?: string
+  onPress: () => void
+  eventName?: AnalyticsEventType
+  eventProperties?: AnalyticsPropertiesList[AnalyticsEventType]
+  style?: StyleProp<ViewStyle>
+}
+
+type WrapperProps = CommonProps & {
+  children: JSX.Element
+}
+
+function Wrapper({
+  eventName,
+  onPress,
+  disabled,
+  testID,
+  children,
+  style,
+  eventProperties,
+}: WrapperProps) {
+  const onPressLocal = React.useCallback(() => {
+    if (eventName) {
+      eventProperties
+        ? ValoraAnalytics.track(eventName, eventProperties)
+        : ValoraAnalytics.track(eventName)
+    }
+    onPress()
+  }, [onPress, eventName])
+
+  return (
+    <Touchable
+      disabled={disabled}
+      // testID={`${testID}_wrapper`}
+      onPress={onPressLocal}
+      borderless={true}
+      hitSlop={variables.iconHitslop}
+      style={style}
+    >
+      {children}
+    </Touchable>
+  )
+}
+
+export type TopBarIconButtonProps = CommonProps & {
```

Comment:
Calling this `TopBarIconButtonV2Props` might make more sense so it matches the name of the component.

Reply:
This doesn't need to be exported: [knip run](https://github.com/valora-inc/wallet/actions/runs/8732674157/job/23960065783?pr=5276#step:4:1178).